modules/mauth-common/src/main/java/com/mdsol/mauth/util/MAuthSignatureHelper.java

```diff
@@ -13,25 +15,101 @@
-    logger.debug("Generating Unencrypted Signature");
-    return httpMethod + "\n" + resourceUrl + "\n" + body + "\n" + appUUID.toString() + "\n" + epochTime;
+  /**
+   * Generate string_to_sign for Mauth V1
+   * @param appUUID: app uuid
+   * @param httpMethod: Http_Verb
+   * @param resourceUrl: resource_url_path (no host, port or query string; first "/" is included)
+   * @param requestBody: request body string
+   * @param epochTime: current seconds since Epoch
+   * @return String
+   *   httpMethod + "\n" + resourceUrl + "\n" + requestBody + "\n" + app_uuid + "\n" + epochTime
+   *
+   */
+  public static String generateUnencryptedSignature(UUID appUUID, String httpMethod, String resourceUrl, String requestBody, String epochTime) {
+    logger.debug("Generating String to sign for V1");
+    return httpMethod + "\n" + resourceUrl + "\n" + requestBody + "\n" + appUUID.toString() + "\n" + epochTime;
+  }
+
+  /**
+   * Generate string_to_sign for Mauth V2
+   * @param appUUID: application uuid
+   * @param httpMethod: Http_Verb
+   * @param resourceUrl: resource_url_path (no host, port or query string; first "/" is included)
+   * @param queryParameters: request parameters string
+   * @param requestBody: request body string
+   * @param epochTime: current seconds since Epoch
+   * @return String
+   *   httpMethod + "\n" + resourceUrl + "\n" + requestBody_digest + "\n" + app_uuid + "\n" + epochTime + "\n" + encoded_queryParameters
+   *
+   * @throws MAuthSigningException
+   */
+  public static String generateStringToSignV2(UUID appUUID, String httpMethod, String resourceUrl,
+      String queryParameters, String requestBody, String epochTime) throws MAuthSigningException{
+    logger.debug("Generating String to sign for V2");
+
+    String bodyDigest;
+    String encryptedQueryParams;
+    try {
+      bodyDigest = getHexEncodedDigestedString(requestBody);
+      encryptedQueryParams = generateEncryptedQueryParams(queryParameters);
+    } catch (IOException e) {
+      logger.error("Error generating Unencrypted Signature", e);
+      throw new MAuthSigningException(e);
+    }
+    return httpMethod.toUpperCase() + "\n" + resourceUrl + "\n" + bodyDigest + "\n"
+        + appUUID.toString() + "\n" + epochTime + "\n" + encryptedQueryParams;
+  }
+
+  /**
+   * Generate string_to_sign for Mauth
+   * @param appUUID: app uuid
+   * @param httpMethod: Http_Verb
+   * @param resourceUrl: resource_url_path (no host, port or query string; first "/" is included)
+   * @param queryParameters: request parameters string
+   * @param requestBody: request body string
+   * @param epochTime: current seconds since Epoch
+   * @param mauthVersion: Mauth version (MSW or MSWV2)
+   * @return String
+   *
+   * @throws MAuthSigningException
+   */
+  public static String generateStringToSign(UUID appUUID, String httpMethod, String resourceUrl,
```

Comment:
If V2 is going to be the default I would use name `generateStringToSign` for V2 and rename this one to be `generateStringToSignV1AndV2` and possibly deprecate it so people will use it with caution

Reply:
I am thinking to use generateStringToSign as a generic method to return the string to sign based on the version specified and have updated code to call **generateStringToSign** instead of generateStringToSignV2.